torch/distributions/constraints.py

```diff
@@ -592,5 +617,7 @@ def check(self, value):
 square = _Square()
 symmetric = _Symmetric()
 positive_definite = _PositiveDefinite()
+symmetric_positive_semidefinite = SymmetricPositiveSemidefinite()
```

Comment:
Do you foresee having to separate the checks for symmetric and positive definite matrices? If we do not have a use case, I would suggest that we keep this as `positive_definite` and `positive_semidefinite` with implicit `symmetric` checks. We can later separate this and deprecate the old constraint names, if needed, but open to other suggestions too.

Reply:
Should have a `_` prefixed to these classes here.

Reply:
> Do you foresee having to separate the checks for symmetric and positive definite matrices? If we do not have a use case, I would suggest that we keep this as `positive_definite` and `positive_semidefinite` with implicit `symmetric` checks. We can later separate this and deprecate the old constraint names, if needed, but open to other suggestions too.

I appreciate your review! That would be great.
Maybe someone can combine `constraints.stack(constraints.symmetric, constraints.positive_definite)`
or PyTorch can provide alias at the below of `torch/distributions/constraints.py` like
```
...
symmetric_positive_semidefinite = _Stack([_Symmetric(), _PositiveDefinite()])
...
```
Not sure it will work correctly, though.
I applied the changes.